File: android/app/src/main/java/com/mediadownloader/extractors/sites/TwitterExtractor.kt
package com.mediadownloader.extractors.sites

import com.mediadownloader.config.Config
import com.mediadownloader.extractors.BaseExtractor
import com.mediadownloader.extractors.DownloadTask

class TwitterExtractor : BaseExtractor {
    override fun matches(url: String): Boolean =
        url.contains("twitter.com") || url.contains("x.com")

    override suspend fun extract(url: String): List<DownloadTask> =
        listOf(
            DownloadTask(
                url = url,
                useYtdlp = true,
                formatId = Config.getDefaultFormat()
            )
        )
}

File: src/extractors/sites/twitter_extractor.cpp
#include "twitter_extractor.hpp"
#include "../../config.hpp"

bool TwitterExtractor::matches(
    const std::string& url
)
{
    return
        url.find("twitter.com")
        != std::string::npos
        ||
        url.find("x.com")
        != std::string::npos;
}

std::vector<DownloadTask>
TwitterExtractor::extract(
    const std::string& url
)
{
    DownloadTask task;

    task.url = url;

    task.use_ytdlp = true;

    task.format_id =
        Config::get_default_format();

    return { task };
}
